reference/conanfile/methods/validate_build.rst

```diff
@@ -0,0 +1,33 @@
+.. _reference_conanfile_methods_validate_build:
+
+
+validate_build()
+================
+
+The ``validate_build()`` method is used to verify if a configuration is valid for building a package. It is different
+from the ``validate()`` method that checks if the binary package is "impossible" or invalid for a given configuration.
+
+The ``validate()`` method should do the checks of the settings and options using the ``self.info.settings``
+and ``self.info.options``.
+
+The ``validate_build()`` method has to use always the ``self.settings`` and ``self.options``:
+
+.. code-block:: python
+
+    from conan import ConanFile
+    from conan.errors import ConanInvalidConfiguration
+    class myConan(ConanFile):
+        name = "foo"
+        version = "1.0"
+        settings = "os", "arch", "compiler"
+        def package_id(self):
+            # For this package, it doesn't matter the compiler used for the binary package
+            del self.info.settings.compiler
```

Comment:
safe_rm?

Reply:
Not necessary here, ``compiler`` will always be there, it was declared as settings.